core/app/models/spree/address.rb

```diff
@@ -174,11 +173,25 @@ def country_iso=(iso)
     end
 
     def country_iso
-      country && country.iso
+      country&.iso
+    end
+
+    def firstname=(value)
+      super
+      update_name
+    end
+
+    def lastname=(value)
+      super
+      update_name
     end
 
     private
 
+    def update_name
```

Comment:
I'd use another name for this method. `update_name` makes me think that an update (save) will be performed while we are only setting the attribute here relying on something else for persisting it. Also, maybe we should set the name only if it is empty?

Reply:
In a following commit I added that check https://github.com/solidusio/solidus/pull/3447/commits/93db0be1d74d5666145d6f0e7f6a052a98c06e28#diff-14b461f472ff6adbfd491ed5674e4e3aR203.
I agree, the `update_name` could be misleading, maybe `fill_name` would be better 🤔 ?